extensions/wikia/PortabilityDashboard/SpecialPortabilityDashboardController.php

```diff
@@ -14,53 +15,55 @@ public function __construct() {
 	}
 
 	public function index() {
+		global $wgBlankImgUrl;
+
 		$model = new PortabilityDashboardModel();
-		$list = $model->getList();
-		$langFilter = $this->getVal( self::LANGUAGE_FILTER_QS_PARAM, '' );
+		$langFilter = $this->getVal( static::LANGUAGE_FILTER_QS_PARAM, '' );
+		$wikiUrlParam = $this->getVal( static::WIKI_URL_QS_PARAM, '' );
 		$isLangFilterSet = !empty( $langFilter );
+		$isWikiUrlParamSet = !empty( $wikiUrlParam );
+		$noResultsInfoMessage = wfMessage(
+			$isLangFilterSet ? 'portability-dashboard-no-results-for-lang-info' : 'portability-dashboard-no-results-for-url-info',
+			PortabilityDashboardModel::WIKIS_LIMIT
+		)->text();
 
-		// template model 
-		$this->response->setVal( 'list', $isLangFilterSet ? $this->filterListByLang( $list, $langFilter ) : $list );
-
-		// template helpers
-		$this->response->setVal(
-			'typelessTemplatesInsightsPath', self::SPECIAL_INSIGHTS_PATH . self::SPECIAL_INSIGHTS_TYPELESS_TEMPLATE_PAGE
-		);
-		$this->response->setVal(
-			'customInfoboxesInsightsPath', self::SPECIAL_INSIGHTS_PATH . self::SPECIAL_CUSTOM_INFOBOXES_PAGE
-		);
-		$this->response->setVal(
-			'langList',
-			$this->extendLanguagesListWithActiveLanguage(
-				self::SUPPORTED_LANGUAGE_FILTERS,
-				$langFilter
-			)
-		);
-		$this->response->setVal( 'isLangFilterSet', $isLangFilterSet );
-		$this->response->setVal( 'langQSParam', self::LANGUAGE_FILTER_QS_PARAM );
-
-		// i18n template strings
-		$this->response->setVal( 'langFilterLabel', wfMessage( 'portability-dashboard-language-filter-label' )->text() );
-		$this->response->setVal( 'dashboardLegend', wfMessage( 'portability-dashboard-hover-info' )->text() );
-		$this->response->setVal( 'allLangFilter', wfMessage( 'portability-dashboard-language-filter-all' )->text() );
-		$this->response->setVal( 'dashboardLabels', $this->getDashboardLabels() );
-		$this->response->setVal(
-			'templatesWithoutTypeUrlTitle', wfMessage( 'portability-dashboard-special-insights-template-without-title'
-		)->text() );
-		$this->response->setVal(
-			'customInfoboxesInsightsUrlTitle', wfMessage( 'portability-dashboard-special-insights-custom-infobox-title'
-		)->text() );
-		$this->response->setVal( 'refreshFreqInfo', wfMessage( 'portability-dashboard-refresh-frequency-info' )->text() );
-		$this->response->setVal( 'noResultsInfo', wfMessage( 'portability-dashboard-no-results-info',
-			PortabilityDashboardModel::WIKIS_LIMIT )->text() );
+		if ( $isWikiUrlParamSet ) {
+			$list = $model->getWikiDataByUrl( Sanitizer::encodeAttribute( $wikiUrlParam ) );
+		} else {
+			$modelList = $model->getRowList();
+			$list = $isLangFilterSet ? $this->filterListByLang( $modelList, $langFilter ) : $modelList;
```

Comment:
I wonder if it can be moved outside of this if-else block along with https://github.com/Wikia/app/pull/11133/files#diff-687fc361036c5b5b06951c4186cbb8f0R26 - technically, it's possible to use URL param to filter by both language and url